contracts/core/liquidators/twap-impl/TWAPAuction.sol

```diff
@@ -39,4 +56,357 @@ contract TWAPAuction is TwoAssetPriceBoundedLinearAuction {
+    }
+
+    /* ============ Internal Helper Functions ============ */
+
+    /**
+     * Resets state for next chunk auction (except minimumBid and token flow arrays)
+     *
+     * @param _twapAuction                  TWAPAuction State object
+     * @param _chunkAuctionSize             Size of next chunk auction
+     */
+    function overwriteChunkAuctionState(
+        TWAPState storage _twapAuction,
+        uint256 _chunkAuctionSize
+    )
+        internal
+    {
+        _twapAuction.chunkAuction.auction.startingCurrentSets = _chunkAuctionSize;
+        _twapAuction.chunkAuction.auction.remainingCurrentSets = _chunkAuctionSize;
+        _twapAuction.chunkAuction.auction.startTime = block.timestamp;
+        _twapAuction.chunkAuction.endTime = block.timestamp.add(auctionPeriod);
+
+        // Since currentSet and nextSet param is not used in calculating start and end price on
+        // TwoAssetPriceBoundedLinearAuction we can pass in zero addresses
+        _twapAuction.chunkAuction.startPrice = calculateStartPrice(
+            _twapAuction.chunkAuction.auction,
+            ISetToken(address(0)),
+            ISetToken(address(0))
+        );
+        _twapAuction.chunkAuction.endPrice = calculateEndPrice(
+            _twapAuction.chunkAuction.auction,
+            ISetToken(address(0)),
+            ISetToken(address(0))
+        );
+    }
+
+    function validateLiquidatorData(
+        ISetToken _currentSet,
+        ISetToken _nextSet,
+        uint256 _startingCurrentSetQuantity,
+        TWAPLiquidatorData memory _liquidatorData
+    )
+        internal
+        view
+    {
+        uint256 rebalanceVolume = LiquidatorUtils.calculateRebalanceVolume(
+            _currentSet,
+            _nextSet,
+            oracleWhiteList,
+            _startingCurrentSetQuantity
+        );
+
+        bytes32 assetPairHash = getAssetPairHashFromCollateral(_currentSet, _nextSet);
+
+        validateTWAPParameters(
+            _liquidatorData,
+            assetPairHash,
+            rebalanceVolume
+        );
+    }
+
+    /**
+     * Validates passed in parameters for TWAP auction
+     *
+     * @param _twapLiquidatorData       Struct of passed liquidator data
+     * @param _assetPairHash            Hash of asset pair being rebalanceds
+     * @param _rebalanceVolume          Value of collateral being rebalanced
+     */
+    function validateTWAPParameters(
+        TWAPLiquidatorData memory _twapLiquidatorData,
+        bytes32 _assetPairHash,
+        uint256 _rebalanceVolume
+    )
+        internal
+        view
+    {
+        require(
+            chunkSizeWhiteList[_assetPairHash].isWithin(_twapLiquidatorData.usdChunkSize),
+            "TWAPAuction.validateTWAPParameters: Passed chunk size must be between bounds."
+        );
+
+        uint256 chunkAuctions = _rebalanceVolume.divCeil(_twapLiquidatorData.usdChunkSize);
+        uint256 expectedTWAPAuctionTime = chunkAuctions.mul(expectedChunkAuctionLength)
```

Comment:
Name suggestion: `expectedRebalanceLength`

Reply:
Would add a comment to describe the intent of the math

Reply:
Why is that name any clearer?